MetaMask/metamask-extension
# fix: IPFS NFTs fetching

## **Description**

Problem: With `IPFS gateway=on` and `Display NFT media=on`, manually imported IPFS NFT render blank.

Diagnosis: `NftController` checks  [`isIpfsGatewayEnabled`](https://github.com/MetaMask/core/blob/ac18b9c8eca7a4aab354bcf0bea82266ceb19a77/packages/assets-controllers/src/NftController.ts#L462), but this value was always undefined even with the setting enabled in extension.  I'm not sure why the field was present in core's preferences controller but not extension's.

Solution: Update extension to toggle `isIpfsGatewayEnabled` preference

Before:

https://github.com/MetaMask/metamask-extension/assets/3500406/d56a426d-83f9-4690-b3a6-85ae854a76ff

After: 

https://github.com/MetaMask/metamask-extension/assets/3500406/68716a12-2da8-4bac-bca0-a87aca88b5df

## **Related issues**

Fixes:

## **Manual testing steps**

1. Set `IPFS gateway=on` and `Display NFT media=on` 
2.  manually import an IPFS NFT (e.g. `0xee79a3e8aef1109a6ee82bf399ce9e1bd43cf5c4` on polygon)
3. it should render

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
